server.coffee

```diff
@@ -194,6 +196,12 @@ extendPublish (name, publishFunction, options) ->
       while handles.length
         handle = handles.shift()
         handle?.stop()
+      # Stop computations in case a subscription is stopped before ready on the client and
+      # the publish is using meteorhacks:unblock's "this.unblock()". Without this, observers
+      # created from cursors inside the autorun will exist forever.
+      while trackedComputations.length
+        computation = trackedComputations.shift()
+        computation?.stop()
```

Comment:
This is strange that this helps you, because `handle` is the same as `computation` and I already `stop` them? So the whole idea why I collect `handle`s there is to clean them up.

Reply:
Maybe it is not the same, but then this is very strange and very problematic because it might point to a bug in server-side tracker implementation.

Reply:
Well your current implementation works only if a computation is returned from the autorun, but not if a cursor is.

Given meteorhacks:unblock is the only way to reproduce this, it seems like a race condition but I really don't know.

Reply:
> Well your current implementation works only if a computation is returned from the autorun, but not if a cursor is.
No, result of calling `Tracker.autorun` is always a handle/computation of this autorun. It does not matter what internally is returned.

Reply:
Also, not sure why you want to use `unblock`, BTW. There are some [strange side effects](https://github.com/meteorhacks/unblock/issues/11#issuecomment-266263478) and [it uses a global variable](https://github.com/meteorhacks/unblock/blob/master/lib/unblock.js#L1) I am pretty scared of seeing.